neo4j-app/neo4j_app/app/config.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+
+import configparser
+import functools
+import io
+from copy import copy
+from typing import Optional, TextIO
+
+from pydantic import Field
+
+from neo4j_app import AppConfig
+from neo4j_app.core.utils.pydantic import (
+    BaseICIJModel,
+)
+from neo4j_app.icij_worker import WorkerConfig, WorkerType
+
+_SHARED_WITH_NEO4J_WORKER_CONFIG = [
+    "neo4j_connection_timeout",
+    "neo4j_host",
+    "neo4j_password",
+    "neo4j_port",
+    "neo4j_uri_scheme",
+    "neo4j_user",
+]
+
+
+class ServiceConfig(AppConfig):
+    neo4j_app_async_dependencies: Optional[str] = "neo4j_app.tasks.WORKER_LIFESPAN_DEPS"
+    neo4j_app_async_app: str = "neo4j_app.tasks.app"
+    neo4j_app_gunicorn_workers: int = 1
+    neo4j_app_host: str = "127.0.0.1"
+    neo4j_app_n_async_workers: int = 1
+    neo4j_app_name: str = "🕸 neo4j app"
+    neo4j_app_port: int = 8080
+    neo4j_app_task_queue_size: int = 2
+    neo4j_app_worker_type: WorkerType = WorkerType.neo4j
+    test: bool = False
+
+    @functools.cached_property
+    def doc_app_name(self) -> str:
+        return self.neo4j_app_name
+
+    def to_worker_config(self, **kwargs) -> WorkerConfig:
+        kwargs = copy(kwargs)
+        # TODO: pass other parameters here
+        if "cancelled_tasks_refresh_interval_s" not in kwargs:
+            kwargs[
+                "cancelled_tasks_refresh_interval_s"
+            ] = self.neo4j_app_cancelled_task_refresh_interval_s
+        if "task_queue_poll_interval_s" not in kwargs:
+            kwargs[
+                "task_queue_poll_interval_s"
+            ] = self.neo4j_app_cancelled_task_refresh_interval_s
+        if "log_level" not in kwargs:
+            kwargs["log_level"] = self.neo4j_app_log_level
+
+        if self.test:
+            from neo4j_app.tests.icij_worker.conftest import MockWorkerConfig
+
+            return MockWorkerConfig(**kwargs)
```

Comment:
I'm a bit surprised to see test code here. You couldn't mock it differently?

Reply:
Yes that's an old piece of code that I introduced at the very beginning, of async workers.
But yes you are right, this is just a matter of adding the HTTP dependency injection definition inside the `ServiceConfig`.

This way we can use `create_app(test_config)` for tests and this will create all test dependencies